conans/client/tools/system_pm.py

```diff
@@ -89,10 +73,20 @@ def install(self, packages, update=True, force=False):
         '"""
         packages = [packages] if isinstance(packages, str) else list(packages)
 
-        if not force and self._installed(packages):
-            return
+        mode = self._get_sysrequire_mode()
 
-        if not self._check_sysrequires_mode(packages):
+        if mode in ("verify", "disabled"):
+            # Report to output packages need to be installed
+            _global_output.warn("The following packages need to be installed:\n %s" % "\n".join(packages))
+            if mode == "disabled":
+                return
+
+            if mode == "verify" and not self._installed(packages):
```

Comment:
I think there is a big bug (not yours) in the `def _installed(self, packages):` function, it is returning True if any of the packages are installed. Let's add a couple of tests to verify that it is working as expected, also when "not packages" it should return True (nothing to install).

Reply:
@memsharded it could be breaking, but it is a bug, right? It is broken and inherited since 2015
```
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 705)     def _installed(self, packages):
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 706)         for pkg in packages:
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 707)             if self._tool.installed(pkg):
ff5fc105 (Luis Martinez de Bartolome Izquierdo 2017-06-09 13:55:08 +0200 708)                 _global_output.info("Package already installed: %s" % pkg)
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 709)                 return True
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 710)         return False
2d2215d9 (Sixten Hilborn                       2017-02-15 22:59:51 +0100 711) 
```

Reply:
Yes you are right!!! I'll fix it!!!

Reply:
The thing is that such ``packages`` is not intended to install different packages, but to provide alternative package names for the desired thing to install. This was contributed by users, but I recall that was the intention. So it works if it finds or installs any of the items in the list.